### ISSUE TITLE
Cartan decomposition does not work for random unitaries

### ISSUE THREAD

**Issue by goerz** [qucontrol/weylchamber#1]
The Cartan decomposition works for "standard gates" but fails for random unitaries, via this test:
https://github.com/qucontrol/weylchamber/blob/43cf5031b914b9439bee0fce466025b71a0011a3/tests/test_cartan_decomposition.py#L10

@danielreich When we wrote this routine back in the day, I remember us testing this rather extensively. So, I don't know why it's failing now. It's in no way urgent to fix this, but at some point we should look into this again.

**Comment by pranavm1502** [comment]
Hi,
Firstly, thank you for making this amazing package so user-friendly!

Is the Cartan decomposition of arbitrary gates, still an unfixed issue?

**Comment by goerz** [comment]
Yes, it's still unfixed. If you're actively using Cartan decomposition and you find that you're affected by this bug, you might be able to find out what's wrong here. Neither me nor @danielreich are currently using this code in any active research, so we'll probably not be able to look into this in the short term.

@Basilewitsch: Do you have any ideas, maybe?
